x-pack/plugins/cases/server/services/cases/index.ts

```diff
@@ -245,107 +245,71 @@ export class CasesService {
     };
   }
 
-  /**
-   * Retrieves the number of cases that exist with a given status (open, closed, etc).
-   */
-  public async findCaseStatusStats({
-    unsecuredSavedObjectsClient,
-    caseOptions,
-    ensureSavedObjectsAreAuthorized,
-  }: {
-    unsecuredSavedObjectsClient: SavedObjectsClientContract;
-    caseOptions: SavedObjectFindOptionsKueryNode;
-    ensureSavedObjectsAreAuthorized: EnsureSOAuthCallback;
-  }): Promise<number> {
-    const cases = await this.findCases({
-      unsecuredSavedObjectsClient,
-      options: {
-        ...caseOptions,
-        page: 1,
-        perPage: MAX_DOCS_PER_PAGE,
-      },
-    });
-
-    // make sure that the retrieved cases were correctly filtered by owner
-    ensureSavedObjectsAreAuthorized(
-      cases.saved_objects.map((caseInfo) => ({ id: caseInfo.id, owner: caseInfo.attributes.owner }))
-    );
-
-    return cases.saved_objects.length;
-  }
-
-  /**
-   * Returns the number of total comments and alerts for a case
-   */
-  public async getCaseCommentStats({
-    unsecuredSavedObjectsClient,
-    ids,
+  public async getCaseStatusStats({
+    searchOptions,
   }: {
-    unsecuredSavedObjectsClient: SavedObjectsClientContract;
-    ids: string[];
-  }): Promise<CaseCommentStats> {
-    if (ids.length <= 0) {
-      return {
-        commentTotals: new Map<string, number>(),
-        alertTotals: new Map<string, number>(),
-      };
-    }
-
-    const getCommentsMapper = async (id: string) =>
-      this.getAllCaseComments({
-        unsecuredSavedObjectsClient,
-        id,
-        options: { page: 1, perPage: 1 },
-      });
-
-    // Ensuring we don't do too many concurrent get running.
-    const allComments = await pMap(ids, getCommentsMapper, {
-      concurrency: MAX_CONCURRENT_SEARCHES,
-    });
-
-    const alerts = await this.getAllCaseComments({
-      unsecuredSavedObjectsClient,
-      id: ids,
-      options: {
-        filter: nodeBuilder.is(`${CASE_COMMENT_SAVED_OBJECT}.attributes.type`, CommentType.alert),
+    searchOptions: SavedObjectFindOptionsKueryNode;
+  }): Promise<{
+    // When I try to do `[CaseStatuses[in-progress]]: number TS gives me a:
+    // A computed property name in a type literal must refer to an expression whose type is a literal type or a 'unique symbol' type.ts(1170)
+    // error
+    [status in CaseStatuses]: number;
```

Comment:
TS is correct on that 🙂. You can do `Promise<Record<CaseStatuses, number>>` which is basically the same with `[status in CaseStatuses]: number`

Reply:
I'll remove the comment 👍